caffe2/operators/hip/pool_op_miopen.hip

```diff
@@ -151,9 +163,11 @@ class MIOPENPoolGradientOp : public ConvPoolOpBase<HIPContext> {
     MIOPEN_ENFORCE(miopenCreateTensorDescriptor(&top_desc_));
     MIOPEN_ENFORCE(miopenCreatePoolingDescriptor(&pooling_desc_));
 
-    if (operator_def.type().substr(0, 7) == "MaxPool") {
+    if ((operator_def.type().substr(0, 7) == "MaxPool") ||
+       (operator_def.type().substr(0, 7) == "MaxPool3D")) {
```

Comment:
Please review all substr() lengths.  It seems they don't match up with the strings we are comparing against.